makefile

```diff
@@ -225,34 +241,48 @@ ASM_FLAGS += ${COMMON_FLAGS}
 
 all: $(APP_OUT_PATH)/$(PROJECT).bin # $(APP_OUT_PATH)/$(PROJECT).hex size
 
-
-.s.o:
+$(OBJ_PATH)/%.o: ../%.s
 	+@$(call MAKE_DIR,$(dir $@))
 	+@echo "Assemble: $(notdir $<)"
-	@$(AS) -c $(ASM_FLAGS) -o $@ $<
+	@$(AS) -c $(ASM_FLAGS) $< -o $@
 
-.S.o:
+$(OBJ_PATH)/%.o: ../%.S
 	+@$(call MAKE_DIR,$(dir $@))
 	+@echo "Assemble: $(notdir $<)"
-	@$(AS) -c $(ASM_FLAGS) -o $@ $<
+	@$(AS) -c $(ASM_FLAGS) $< -o $@
+
+$(OBJ_PATH)/%.o: ../%.c
+	+@$(call MAKE_DIR,$(dir $@))
+	+@echo "Compile: $(notdir $<)"
+	@$(CC) $(C_FLAGS) $(INCLUDE_PATHS) $< -o $@
+%.o: C_FLAGS += ${THIRD_PARTY_COMMON_FLAGS}
```

Comment:
I don't really follow here, how do the third party flags get applied and to what? Everything except the UWRT objects? Sorry it's been a very long time since I touched make lol.

Reply:
when creating the list of objects, i also created a seperate like of "uwrt objects" which includes all the sources in `app` and `lib`(aka everything we wrote). Since make always matches the rule with the longest matching stem, those "uwrt" source files use either `$(UWRT_C_OBJ): $(OBJ_PATH)/%.o: ../%.c` or  `$(UWRT_CPP_OBJECTS): $(OBJ_PATH)/%.o: ../%.cpp` depending on whether theyre a c or cpp file.

`$(UWRT_C_OBJ): C_FLAGS += ${UWRT_COMMON_FLAGS}` and the other lines similar to this modify the rules listed above so that their flag variables include extra flags (either `UWRT_COMMON_FLAGS` or `THIRD_PARTY_COMMON_FLAGS`).

Any file that doesnt match the uwrt rules, will use the more general rules. the general rules apply the third party flags.

Something of note is that make will apply these extra "rule specific variables` whether its before the rule definition or after. To maybe make this a little more intuitive to those not well versed in make (aka me), I've moved the flag modification line before the rule definition.

flags have been tested to work